### DISCUSSION TITLE
Support for Web open3d visualizer

### DISCUSSION THREAD

**Discussion by Shaamsi** [dfki-ric/pytransform3d#160] [Ideas]
Can pytransform3d support web visualizer by open3d:
you just need to add this [here:](https://github.com/rock-learning/pytransform3d/blob/master/pytransform3d/visualizer/_figure.py)
o3d.visualization.webrtc_server.enable_webrtc()

tutorial [here](http://www.open3d.org/docs/release/tutorial/visualization/web_visualizer.html).

so we do not have to do fig,show() but use the browser, is it possible to have it?

**Comment by AlexanderFabisch** [comment]
Hi @Shaamsi , I looked into it and I guess it is a bit more complicated. The Figure class does not use the open3d.visualization interface. It uses the [open3d.visualization.Visualizer](http://www.open3d.org/docs/release/python_api/open3d.visualization.Visualizer.html#open3d.visualization.Visualizer) class (and VisualizerWithKeyCallback).

If I read the patch notes for Open3D 0.13 correctly, the Visualizer class will also be deprecated in Open3D 0.14, so I have to make a transition to the new Open3D viewer as well. Furthermore, I was planning to offer the OffscreenRenderer as a backend (https://github.com/rock-learning/pytransform3d/issues/95). It is currently a bit unpredictable how we can make all these things work with the same interface.

If you have any suggestions, let me know!

I also don't plan to use the web viewer at the moment, so the time that I can invest in it is limited.
